### PR TITLE
Scheduler.start: check port file exists at start

### PR DESCRIPTION
**Pull request by matthewrmshin**
And after configuring Pyro daemon. This should make sure that we cannot run 2
suites simultaneously on the same directory by accident.

The problem this fixes can be serious after #1847.

@hjoliver @arjclark please review.

### CODE REVIEW

File: lib/cylc/scheduler.py [hunk 1]
```diff
@@ -203,28 +203,25 @@ def __init__(self, is_restart, options, args):
 
     def start(self):
         """Start the server."""
+        self._check_port_file_exists(self.suite)
         self._print_blurb()
-        try:
```
Comment (matthewrmshin) [inline lib/cylc/scheduler.py]: This `try...except` block is a bit pointless, as this method is caught by a similar block in the `main` function.

File: lib/cylc/scheduler.py [hunk 2]
```diff
@@ -203,28 +203,25 @@ def __init__(self, is_restart, options, args):
 
     def start(self):
         """Start the server."""
+        self._check_port_file_exists(self.suite)
```
Comment (matthewrmshin) [inline lib/cylc/scheduler.py]: Check port file 1st time.

File: lib/cylc/scheduler.py [hunk 3]
```diff
@@ -309,6 +336,21 @@ def _print_blurb():
             print logo_lines[i], ('{0: ^%s}' % lmax).format(license_lines[i])
         print
 
+    def _configure_pyro(self):
+        """Create and configure Pyro daemon."""
+        self.pyro = PyroDaemon(self.suite, self.suite_dir)
+        self.port = self.pyro.get_port()
+        port_file_path = self._check_port_file_exists(self.suite)
```
Comment (matthewrmshin) [inline lib/cylc/scheduler.py]: Check port file 2nd time.

**Comment by hjoliver** [comment]
@matthewrmshin -

>  This should make sure that we cannot run 2 suites simultaneously on the same directory by accident.

I thought we already had this covered with the port file check - what's the difference in functionality here exactly?

**Comment by hjoliver** [comment]
OK I think I understand now.  On master, an attempted start of a second suite instance fails with "ERROR, port file exists", **but** it nukes the port file for the first suite.

The issue description doesn't say this, or mention that it is a result of #1847 though (since last release) - if you agree with this, could you amend the description please ?

File: lib/cylc/scheduler.py [hunk 4]
```diff
@@ -277,6 +274,36 @@ def start(self):
             stats.sort_stats('cumulative')
             stats.print_stats()
             print string_stream.getvalue()
+        print
+
+    @staticmethod
+    def _check_port_file_exists(suite):
```
Comment (hjoliver) [inline lib/cylc/scheduler.py]: This method should probably be called `_check_port_file_does_not_exist()`, no?

File: lib/cylc/scheduler.py (hunk 3 above)
Comment (hjoliver) [inline lib/cylc/scheduler.py]: Is this to guard against another suite instance starting up in between the first port file check and configuring pyro?

**Comment by hjoliver** [comment]
Review 1: tests all pass, a couple of minor questions but otherwise good.  Over to @arjclark

**Comment by matthewrmshin** [comment]
> The issue description doesn't say this, or mention that it is a result of #1847 (recent, since last release) - if you agree with this, could you amend the description please ?

@hjoliver This is now mentioned.

File: lib/cylc/scheduler.py (hunk 4 above)
Comment (matthewrmshin) [inline lib/cylc/scheduler.py]: Changed as suggested.

**Comment by matthewrmshin** [comment]
Branch re-based.

**Comment by arjclark** [comment]
Looks ok as far as I can tell. Test battery passing in my environment.

**Pull request merged**